file-formats/nrob/nrob.json

```diff
@@ -0,0 +1,157 @@
+          "title": "Percent Warning",
+          "description": "Percentage of numbers remaining in a number range, upon reaching which in number assignment a warning is given. It must be between 0.1 and 99.9.\n\nExample: You have defined an interval from 1 to 1000. If you want to issue a
+ warning at the number 900, enter 10 (%) here.",
+          "type": "number",
+          "minimum": -9999.9,
+          "maximum": 9999.9
+        },
+        "subType": {
+          "title": "Sub Type",
+          "description": "If you want to create subobjects for the elements of a field of the application table, specify the relevant data element for this table field. This data element must be active in the Data Dictionary and must have a check tab
+le. The domain of the data element must have a field length of between 1 and 6.",
+          "type": "string",
+          "maxLength": 30
+        },
+        "untilYear": {
+          "title": "Until Year",
+          "description": "If you want the records of the business object to be differentiated by financial year, you set this option true. This structures the intervals of a number range object.",
+          "type": "boolean"
+        },
+        "rolling": {
+          "title": "Rolling",
+          "description": "For each interval of a number range object, the system checks when the available characters are used up. In the true setting, once the available characters of an interval are used up, the system starts again from the beginni
+ng and the lowest number is assigned again. You can suppress this behavior by setting this option false. If you do this, no more characters from an interval are assigned once the characters of the interval are used up. You can suppress rollover for t
+he entire number range object, that is, the setting applies to all the intervals contained in the object.",
+          "type": "boolean"
+        },
+        "prefix": {
+          "title": "Prefix",
+          "description": "If it is set to true determined numbers consist of the prefix (name of subobject) and the numbers.",
+          "type": "boolean"
+        }
+      },
+      "additionalProperties": false,
+      "required": [
+        "numberLengthDomain",
+        "percentage",
+        "subType",
+        "untilYear",
+        "rolling",
+        "prefix"
+      ]
+    },
+    "configuration": {
+      "title": "Configuration",
+      "description": "Configuration-specific fields",
+      "type": "object",
+      "properties": {
+        "transactionId": {
+          "title": "Transaction ID",
+          "description": "Transaction code for application specific transaction",
+          "type": "string",
+          "maxLength": 20
+        },
+        "buffering": {
+          "title": "Buffering",
+          "description": "Choose a buffer type for no buffering, for buffering via main memory or for parallel buffering. Default is mainBuffer.",
+          "type": "string",
+          "enum": [
+            "mainBuffer",
```

Comment:
Do we need "default" in the schema here?
In the previous schema there was "default" : "mainBuffer" which we currently can not print to the schema because we don't have an abap doc annotation for that

Reply:
We have to double-check whether we need it for the JSON editor use case.

Reply:
We are adding an abap doc annotation "$default" to our generator. To the simple transformation we write that this field should not be written when it has this default value